apps/api/tests/test_definitions.py

```diff
@@ -113,6 +117,41 @@ def test_instructor_user_can_get_and_post_on_definitions(self):
         )
         assert resp.status_code == 200
 
-        resp = self.client.delete(reverse('api:definition-detail', kwargs={'version': 'v1', 'pk': d_pk}))
+        resp = self.client.delete(reverse('api:definition-detail', kwargs={'version': 'v1', 'pk': definition_pk}))
         assert resp.status_code == 204
 
+    def test_second_instructor_user_can_not_put_on_first_instructor_definitions(self):
+        self.client.login(username='user123', password='pass123')
+
+        resp = self.client.get(reverse('api:definition-list', kwargs={'version': 'v1'}))
+        assert resp.status_code == 200
+
+        resp = self.client.post(
+            reverse('api:definition-list', kwargs={'version': 'v1'}),
+            data={
+                'klass': self.klass.pk,
+                'creator': self.instructor.pk,
+                'due_date': timezone.now(),
+                'name': 'test_definition',
+                'description': 'test'
+                  }
+        )
+        definition_pk = resp.json()['id']
+        data = resp.json()
+        assert data['name'] == 'test_definition'
+        assert resp.status_code == 201
```

Comment:
Rather than using the API for definition creation, let's use a factory here. That way, we're not testing the creation and listing API while we only want to test updating and deleting.

Reply:
Fixed a lot here using setup creators now